onnxruntime/core/providers/cuda/cuda_execution_provider.cc

```diff
@@ -2442,6 +2444,15 @@ static bool ConvTransposeNeedFallbackToCPU(const onnxruntime::Node& node, const
     }
   }
 
+#ifdef ENABLE_CUDA_NHWC_OPS
+  if (prefer_nhwc) {
+    // NHWC implementation doesn't handle transpose of W if it's not an initializer
+    if (!graph_viewer.IsConstantInitializer(node.InputDefs()[1]->Name(), true)) {
+      return true;
+    }
```

Comment:
IMO the transpose for inputs should be added in the layout transformer. Instead of simply transforming the first input (if not a `Resize`) the transformer should check if the input is a constant and depending on that add a transpose node to the input. 
https://github.com/microsoft/onnxruntime/blob/main/onnxruntime/core/optimizer/layout_transformation/layout_transformation.cc#L169-L193

I guess the fix for 1.18 is fine with simply dropping such nodes to CPU, please correct me if my understanding is wrong.

Reply:
Ideally the conversion of ConvTranspose to NHWC would add a transpose for W, but not sure that's possible here. Different EPs are transposing the W in different ways so the input shape for the NHWC ConvTranspose would differ. e.g. XNNPACK splits the C dimension by group (rank increases by 1), and moves the per-group C values to the end vs. CUDA EP that moves the M/group values to the end. 

Should be possible for Conv though. AFAIK all EPs move the C/group dim of W to the end.

Transpose of other inputs than the first would be operator specific as you need to differentiate inputs that have channels data from those that don't. e.g. bias input. Inferring from the rank is one way, but there may be use cases where that approach is insufficient.